### ISSUE TITLE
Units in axis labels

### ISSUE THREAD

**Issue by aplavin** [JuliaPlots/Makie.jl#3890]
### Feature description

I didn't find an option to put units into axis labels instead of tick labels. Is it available anywhere?
Maybe that should also be the default? I very rarely see plots with units written at each tick label in practice, the most common way is to put them into axis labels.

### For plot types, please add an image of how it should look like

Top: current Unitful support.
Bottom: how I'd really like it to look.
![image](https://github.com/MakieOrg/Makie.jl/assets/687995/e8e5d16a-7ce8-4ba0-8a3b-d68ba05c92e9)

**Comment by DanDeepPhase** [comment]
I found this in the code: https://github.com/MakieOrg/Makie.jl/blob/e73ea0ea569bf2b1b08061d03e769eb301acdafa/src/dim-converts/unitful-integration.jl#L137
The example is a little wrong, because it hides units for Y and then labels them in X, but this works for your example:

```julia
f=Figure()
ax=Axis(f[1,1], 
    dim1_conversion=Makie.UnitfulConversion(u"m"; units_in_label=false),
    xlabel="Distance (m)",
    dim2_conversion=Makie.UnitfulConversion(u"W"; units_in_label=false),
    ylabel="Power (W)",)
mak = scatter!(ax,(1:5)u"m", (10:10:50)u"W")
```
Not sure if there is a simpler way. It does retain the units in the axis, so following up with same dimension units works.
```julia
mak2 = scatter!(ax, (100:100:500)u"cm", (0.005:0.010:0.045)kW)
```

**Comment by icweaver** [comment]
Just saw this really nice approach for tick label units in AoG

https://github.com/MakieOrg/AlgebraOfGraphics.jl/pull/619

https://aog.makie.org/v0.9.6/examples/scales/units#units

> Wonder what's the reason for making units a part of ticklabels: are there some fields where such plots are more common than those with units in axislabels?

I am wondering this too. From the PR, it seems that the design choice for per-tick unit labels was made because it works better with the Observable update pipeline?

**Comment by SimonDanisch** [comment]
I'm pretty sure I initially implemented them to be in the labels and then it got complicated somehow, so when I rushed to finally release it, I had to remove it.
Would be nice to bring it back!

**Comment by icweaver** [comment]
That would be dope! Is your thinking that this would not require figure level recipe support? I see [Julius' suggestion of using SpecAPI](https://github.com/MakieOrg/Makie.jl/issues/4692#issuecomment-2563437387) to potentially get around this, but wasn't sure. I really need to learn how to use it one of these days!

On a sort of related note, I think we would really appreciate your eyes on this [recent DynamicQuantities.jl PR](https://github.com/SymbolicML/DynamicQuantities.jl/pull/165) we have in the works when you have some time, since I'm pretty much just cribbing from your [Unitful.jl PR](https://github.com/MakieOrg/Makie.jl/pull/3226), haha

**Comment by icweaver** [comment]
Dang this is slick

```julia
# DynamicQuantities.jl example
x = (6:10)us"cm/mm"
y = (7:11)us"g"

fig = Figure()
scatter(fig[1, 1], y)
plot(fig[1, 2], x, y)
fig
```

![Image](https://github.com/user-attachments/assets/341fc477-5e06-48f5-b2e5-4e21478e285c)

<details><summary>Details</summary>

```julia
using CairoMakie, DynamicQuantities

import Makie.SpecApi as S

function Makie.convert_arguments(
    ::Type{<:AbstractPlot}, obj::AbstractVector{<:UnionAbstractQuantity}
)
    qa = QuantityArray(obj)
    spec_ax = S.Axis(plots=[S.Lines(qa.value)];
        ylabel = string("Y[ ", qa.dimensions, "]")
    )
    return S.GridLayout(spec_ax)
end

function Makie.convert_arguments(
    ::Type{<:AbstractPlot},
    x::AbstractVector{<:UnionAbstractQuantity},
    y::AbstractVector{<:UnionAbstractQuantity},
)
    qa_x = QuantityArray(x)
    qa_y = QuantityArray(y)
    spec_ax = S.Axis(plots=[S.Lines(qa_x.value, qa_y.value)];
        xlabel = string("X [", qa_x.dimensions, "]"),
        ylabel = string("Y [", qa_y.dimensions, "]")
    )
    return S.GridLayout(spec_ax)
end
```

</details>

I still have a bunch of gaps to cover, but is this the sort of gist of what folks had in mind?

**Comment by icweaver** [comment]
Yep, I just hit that edge recently too. Still learning a bunch, thanks for the `to_plotspec` tip, Alexander!
